.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,29 @@
+name: golangci-lint
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+# Remove all permissions from GITHUB_TOKEN except metadata.
+permissions: {}
+
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        working-directory:
+          - ""
+          - test
+          - hack/tools
+    steps:
+      - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # tag=v3.1.0
+      - uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # tag=v3.3.1
+        with:
+          go-version: 1.19
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@0ad9a0988b3973e851ab0a07adf248ec2e100376 # tag=v3.3.1
+        with:
+          version: v1.50.1
+          working-directory: ${{matrix.working-directory}}
```

Comment:
Do we need a matrix here? Could we just run it over all the source?

Reply:
Yes, then I should remove these lines with 
```yml
strategy:
      matrix:
        working-directory:
          - ""
          - test
          - hack/tools
```

as well right?

Reply:
Lets leave it as is

Reply:
Okay then resolving this